docs/source/specs/openapi.yml

```diff
@@ -729,6 +771,135 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/Error'
+  /api/v1/reports/costs/:
+    get:
+      summary: Query to obtain cost reports
+      operationId: getCostReports
+      tags:
+        - Report
+      parameters:
+        - in: header
+          name: Authorization
+          description: e.g. Token 45138a913da44ab89532bab0352ef84b
+          schema:
+            type: string
+          required: true
+        - in: query
+          name: group_by
+          schema:
+            $ref: '#/components/schemas/ReportGroupingCost'
+          description: The grouping to apply to the cost report. No grouping by default.
+        - in: query
+          name: resolution
+          schema:
+            $ref: '#/components/schemas/ReportResolution'
+          description: The resolution to apply to the cost report. Default resolution is daily.
+        - in: query
+          name: time_scope
+          schema:
+            $ref: '#/components/schemas/ReportTimeScope'
+          description: The time scope to apply to the cost report. Default resolution is '-10', meaning the last 10 days. Last 30 days would be '-30'. '-10' and '-30' are only valid when used with 'daily' resolution.'-1' is used for current month, '-2' is used for last month when resolution is 'monthly' otherwise invalid.
+      responses:
```

Comment:
Since all of these end points are queries, shouldn't we also be returning a 404 or will they always get back a 200 with an empty report?

Reply:
My intent is we would give back 200 with an empty report.

Reply:
ok.  I'll leave it to @dmiller9911 as to how that impacts him.

Reply:
Ideally we would still give back the bucketed time just with 'null' for the values to clarify no data for that time period

Reply:
Should we go ahead and add `nullable: true` as explained [here](https://swagger.io/docs/specification/data-models/data-types/)